src/bundle/Controller/User/ChangePasswordController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUiBundle\Controller\User;
+
+use eZ\Publish\API\Repository\Exceptions\ContentFieldValidationException;
+use eZ\Publish\API\Repository\Exceptions\ContentValidationException;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\API\Repository\LanguageService;
+use EzSystems\EzPlatformAdminUi\Form\Data\User\ChangePasswordData;
+use EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory;
+use EzSystems\EzPlatformAdminUi\Notification\NotificationHandlerInterface;
+use EzSystems\EzPlatformAdminUiBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Exception;
+
+class ChangePasswordController extends Controller
+{
+    /** @var NotificationHandlerInterface */
+    private $notificationHandler;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /** @var LanguageService */
+    private $userService;
+
+    /** @var FormFactory */
+    private $formFactory;
+
+    /** @var TokenStorageInterface */
+    private $tokenStorage;
+
+    /**
+     * @param NotificationHandlerInterface $notificationHandler
+     * @param TranslatorInterface $translator
+     * @param UserService $userService
+     * @param FormFactory $formFactory
+     * @param TokenStorageInterface $tokenStorage
+     */
+    public function __construct(
+        NotificationHandlerInterface $notificationHandler,
+        TranslatorInterface $translator,
+        UserService $userService,
+        FormFactory $formFactory,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->notificationHandler = $notificationHandler;
+        $this->translator = $translator;
+        $this->userService = $userService;
+        $this->formFactory = $formFactory;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Symfony\Component\Translation\Exception\InvalidArgumentException
+     * @throws ContentValidationException
+     * @throws ContentFieldValidationException
+     * @throws InvalidOptionsException
+     */
+    public function changePasswordAction(Request $request): Response
+    {
+        $form = $this->formFactory->changePassword(
```

Comment:
No need to have 3 lines here, also `FormFactory::changePassword` method could initialize `ChangePasswordData` when no arguments are passed. I believe we are already doing it for other forms.